src/Hadolint/Rules.hs

```diff
@@ -44,6 +44,21 @@ type IgnoreRuleParser = Megaparsec.Parsec Void Text.Text
 
 type ParsedFile = [InstructionPos Bash.ParsedBash]
 
+-- | An instructions checker function that gets the current state and a line number.
+-- It should return the new state and whether or not the check passed for the given instruction.
+type SimpleCheckerWithState state
+     = state -> Linenumber -> Instruction Bash.ParsedBash -> (state, Bool)
+
+-- | An instructions checker function that gets the line number.
+-- It should whether or not the check passed for the given instruction.
```

Comment:
I do not understand this line. Can you please explain/rephrase it?

Reply:
Will do :)

Reply:
done!